### PR TITLE
Add alpha plotting from demo

### PR DESCRIPTION
**Pull request by gwarmstrong**
This adds the implementation of an endpoint to return a Vega schema for an Alpha Diversity percentile plot.

This plot is a CDF of the alpha diversities for, with an optional mark for a query sample for a given query.

The plot may look something like below, when rendered on an HTML page with the Vega javascript package:

Screen Shot 2020-07-09 at 12 58 19 PM

### CODE REVIEW

File: microsetta_public_api/api/plotting.py [hunk 1]
```diff
@@ -1,11 +1,76 @@
...
+def plot_alpha_filtered_json_query(body, alpha_metric=None, percentiles=None,
                                    sample_id=None):
-    raise NotImplementedError()
+    repo = MetadataRepo()
+
+    return _plot_alpha_percentiles_querybuilder(alpha_metric, percentiles,
+                                                body, repo, sample_id)
+
+
+def _plot_alpha_percentiles_querybuilder(alpha_metric, percentiles, query,
```
Comment (wasade) [inline microsetta_public_api/api/plotting.py:31]: I think the query matching may make sense to perform higher in the stack? The primary responsibility of this function seems like it's supposed to be plotting, but right now it's doing some filtering, validation of the filtering, and then plotting

File: microsetta_public_api/api/plotting.py [hunk 2]
```diff
@@ -1,11 +1,76 @@
...
+        matching_ids, AlphaRepo, 'available_metrics', alpha_metric, 'metric',
+    )
+    if sample_id:
+        _, error_code, error_response = _filter_matching_ids(
+            [sample_id], AlphaRepo, 'available_metrics', alpha_metric,
+            'metric', error_code=error_code, error_response=error_response,
+        )
+    if error_response:
+        return error_response, error_code
+    if len(matching_ids) <= 1:
```
Comment (wasade) [inline microsetta_public_api/api/plotting.py:44]: for example, this and the error_response check above could be performed upstream of this method which would make this function a lot more focused

File: microsetta_public_api/api/plotting.py (hunk 1 above)
Comment (gwarmstrong) [reply microsetta_public_api/api/plotting.py:31]: Either higher in the stack, or separate this function out into sub-functions. I think either option will make things more readable/organized.
Good catch, thanks!

Comment (wasade) [reply microsetta_public_api/api/plotting.py:31]: good call -- just a decomposition into a separate function would work too and may be better

Comment (gwarmstrong) [reply microsetta_public_api/api/plotting.py:31]: Done in https://github.com/biocore/microsetta-public-api/pull/41/commits/f6ef74820fd21ca9e603dffbd71a6ad4f5c11d08

**Pull request merged**